climpred/toy.py

```diff
@@ -0,0 +1,259 @@
+    noise_amplitude=noise_amplitude,
+    signal_P=signal_P,
+    ramp=ramp,
+):
+    """Create initialized ensemble.
+
+    Args:
+        lead (xr.DataArray): lead times to create synthetic data for.
+        ninit (type): number of initialzations. Defaults to 3.
+        nmember (type): number of members. Defaults to 3.
+        noise_amplitude (type): amplitude of the gaussian noise.
+        ramp (np.array): Ramping up function.
+        signal_amplitude (float): amplitude of the signal.
+        signal_P (float): period of the signal.
+
+    Returns:
+        xr.DataArray: initialized ensemble with potentially predictable variation.
+
+    """
+    # span range of initial conditions
+    init_range = 2 * signal_P
+    to = xr.DataArray(np.random.rand(ninit) * init_range - init_range, dims='init')
+    # create initialized
+    init = signal(
+        lead=lead, lead_offset=to, signal_amplitude=signal_amplitude, signal_P=signal_P
+    ) + create_noise(
+        lead=lead,
+        ninit=ninit,
+        nmember=nmember,
+        noise_amplitude=noise_amplitude,
+        ramp=ramp,
+    )
+    return init
+
+
+def run_skill_for_ensemble(
+    lead=lead,
+    nmember=5,
+    ninit=5,
+    signal_P=signal_P,
+    signal_amplitude=signal_amplitude,
+    noise_amplitude=noise_amplitude,
+    ramp=ramp,
+    bootstrap=10,
+    plot=True,
+    ax=None,
+    label=None,
+    color='k',
+    compute=compute_perfect_model,
+    **metric_kwargs,
+):
+    """Short summary.
+
+    Args:
+        lead (xr.DataArray): lead times to create synthetic data for.
+        ninit (type): number of initialzations. Defaults to 3.
+        nmember (type): number of members. Defaults to 3.
+        noise_amplitude (type): amplitude of the gaussian noise.
+        ramp (np.array): Ramping up function.
+        signal_amplitude (float): amplitude of the signal.
+        signal_P (float): period of the signal.
+        bootstrap (int): bootstrap iterations to calc synthetic skill. Defaults to 10.
+        plot (bool): plot as boxplot. Defaults to True.
+        ax (plt.axes): using existing ax. Defaults to None.
+        label (str): label in legend. Defaults to None.
+        color (str): color of boxplot. Defaults to 'k'.
+        compute (function): compute_perfect_model or compute_hindcast
+        **metric_kwargs (type): Argument to be passed to compute `**metric_kwargs`.
+
+    Returns:
+        whisker_line: to use in legend, if plot.
+        xr.DataArray: skill, else.
+
+    """
+    s = []
+    # for i in tqdm(range(bootstrap),desc='bootstrap'):
+    for i in range(bootstrap):
+        ds = create_initialized(
+            nmember=nmember,
+            ninit=ninit,
+            signal_P=signal_P,
+            signal_amplitude=signal_amplitude,
+            noise_amplitude=noise_amplitude,
+        )
+        s.append(compute_perfect_model(ds, ds, **metric_kwargs))
+    s = xr.concat(s, 'bootstrap')
+    s['lead'] = lead.values
+    if plot:
```

Comment:
Maybe have a separate plotting module

Reply:
Could honestly have this whole thing based on a `toy` class with methods embedded inside, one of which is `.plot()`.